### PR TITLE
Development

### PR DESCRIPTION
**Pull request by hntejas**
Initial PR of dev steam to merge development to main

### CODE REVIEW

File: src/components/VideoPage/video-page.css [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+
+.video-wrapper {
+    position: relative;
+    overflow: hidden;
+    width: 100%;
+    padding-top: 56.25%;
+}
+
+.video-details {
+
```
Comment (nitinmalave) [inline src/components/VideoPage/video-page.css]: Hey @hntejas, please take a look at this extra spaces here.

File: src/data.js [hunk 2]
```diff
@@ -0,0 +1,240 @@
...
+    },
+    thumbnailImgUrl:
+      "https://i.ytimg.com/vi/jx5jmI0UlXU/hqdefault.jpg?sqp=-oaymwEbCKgBEF5IVfKriqkDDggBFQAAiEIYAXABwAEG&rs=AOn4CLBn_nysLus7y_HpRR0bjx1dxmm9TQ",
+  },
+  {
+    id: 9,
+    embedId: "btj35dh3_U8",
+    title: "Callback Functions in JS ft. Event Listeners 🔥",
+    views: 41088,
+    age: "1 Years",
```
Comment (nitinmalave) [inline src/data.js:143]: Hey @hntejas, I am not sure but this property name "age" is not fitting with context i think. I would prefer to have at is "postedAt" or "publishedAt"? It's my opinion and not required to be correct but please think around it as i have implemented it as this.

File: public/index.html [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+	<title>Dev Stream</title>
+</head>
+
+<body>
+	<noscript>
+		You need to enable JavaScript to run this app.
+	</noscript>
+	<div id="root"></div>
+	<!--
+      This HTML file is a template.
```
Comment (nitinmalave) [inline public/index.html]: Please take a look at this file and you can consider to remove the commented code from here.

File: src/components/VideoList/video-list.css [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    margin: 4rem 0;
+  }
+}
+
+
+
+
+
+
+/* @media (max-width: 1200px) {
```
Comment (nitinmalave) [inline src/components/VideoList/video-list.css]: Please take a look at this file and you can consider to remove the commented code from here.

File: src/styles.css [hunk 5]
```diff
@@ -0,0 +1,35 @@
+@import url("https://fonts.googleapis.com/css2?family=Roboto:wght@400;500&display=swap");
+
```
Comment (nitinmalave) [inline src/styles.css:2]: Please take a look at this file and remove extra spaces from in between lines.

File: src/styles.css [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  margin-left: 4rem;
+  margin-right: 4rem;
+}
+
+.section-heading {
+  font-weight: 500;
+  padding-top: 1rem;
+}
+
+.Toastify__progress-bar {
```
Comment (nitinmalave) [inline src/styles.css:25]: Please take a look at this classname, you can keep it consistent with rest of your css classes.

File: src/components/VideoPage/VideoPage.js [hunk 7]
```diff
@@ -0,0 +1,124 @@
...
+    });
+  };
+
+  return (
+    <div className="video-page-container">
+      <div className="video-page-left">
+        <div className="video-wrapper">
+          <iframe
+            className="iframe-video"
+            src={"https://www.youtube.com/embed/" + embedId + "?enablejsapi"}
```
Comment (nitinmalave) [inline src/components/VideoPage/VideoPage.js]: Would like to know why you have appended "?enablejsapi" ahead of your id?

File: src/components/VideoPage/VideoPage.js [hunk 8]
```diff
@@ -0,0 +1,124 @@
...
+  };
+
+  useEffect(() => {
+    userDispatch({
+      type: userActionTypes.UPDATE_HISTORY,
+      payload: {
+        videoEmbedId: embedId,
+      },
+    });
+    window.scroll({
```
Comment (nitinmalave) [inline src/components/VideoPage/VideoPage.js:40]: Would like to know the purpose behind this line?

**Review by nitinmalave — CHANGES_REQUESTED** [review]

File: src/components/VideoPage/VideoPage.js (hunk 8 above)
Comment (hntejas) [reply src/components/VideoPage/VideoPage.js:40]: When the user selects a video from the suggestions in the video page, the redirect is done to the same component and hence view stays at the same position of the page. So when there is a change in the video I'm scrolling to the top to provide a good user experience

File: src/components/VideoPage/VideoPage.js (hunk 7 above)
Comment (hntejas) [reply src/components/VideoPage/VideoPage.js]: This query does not have any impact now and I'll remove it. But the purpose is to take control of the video using JS.

File: src/data.js (hunk 2 above)
Comment (hntejas) [reply src/data.js:143]: postedAt or publishedAt will usually refer to a date of video being posted. But with my current data setup I donot have a date, hence the term "age". Once the backend is setup, I'll be adding the postedAt and will be deriving the age based on that date.

File: src/styles.css (hunk 6 above)
Comment (hntejas) [reply src/styles.css:25]: This classname is from an external library component, I'm overriding it to my use case

File: src/data.js (hunk 2 above)
Comment (nitinmalave) [reply src/data.js:143]: Great idea!

File: src/styles.css (hunk 6 above)
Comment (nitinmalave) [reply src/styles.css:25]: sure.

File: src/components/VideoPage/VideoPage.js (hunk 8 above)
Comment (nitinmalave) [reply src/components/VideoPage/VideoPage.js:40]: Thanks :)

**Comment by nitinmalave** [comment]
LGTM, Review done - GTG ahead.

**Review by kajol-rep — COMMENTED** [review]
Great work everything looks good to me. 
Just few suggestions I would like to give:

1. In history page as you have mentioned recent videos, to follow up this you can display video list in reverse. So the last viewed video will appear at the top.
2. Also, maybe you can rename your folder named store and give it a appropriate name. But if you have a reason behind it then its fine.

**Comment by hntejas** [comment]
> Great work everything looks good to me.
> Just few suggestions I would like to give:
> 
> 1. In history page as you have mentioned recent videos, to follow up this you can display video list in reverse. So the last viewed video will appear at the top.
> 2. Also, maybe you can rename your folder named store and give it a appropriate name. But if you have a reason behind it then its fine.

Hi Kajol,

Thanks for reviewing the code. 
1) I have made the change to show the history in the right order 
2) I have named it store by the convention of Redux concept. In here I'm trying to emulate the redux store using the combination of context and reducer. But I'll surely think of a much developer friendly name for next iterations.

**Comment by kajol-rep** [comment]
> > Great work everything looks good to me.
> > Just few suggestions I would like to give:
> > 
> > 1. In history page as you have mentioned recent videos, to follow up this you can display video list in reverse. So the last viewed video will appear at the top.
> > 2. Also, maybe you can rename your folder named store and give it a appropriate name. But if you have a reason behind it then its fine.
> 
> Hi Kajol,
> 
> Thanks for reviewing the code.
> 
> 1. I have made the change to show the history in the right order
> 2. I have named it store by the convention of Redux concept. In here I'm trying to emulate the redux store using the combination of context and reducer. But I'll surely think of a much developer friendly name for next iterations.

Okay didn't know about this. thanks for the explanation.
Once again your code looks clean and consistent, got to learn lot of things.

**Pull request merged**